Need to add fast math functions

There are many places in our code where we call into slow deterministic gm_xxxx functions from GameMath but that don't need to be deterministic. We should add faster versions of Max, Min, Pow, Cos, Sin, Sqrt, Acos, Asin, Atan, Atan2, Tan, Ceil, Floor, Fast_To_Int_Floor, Fast_To_Int_Ceil, Fast_To_Int_Truncate, Fast_Float_Floor, Fast_Float_Ceil and Lrintf and use those in places where we are confident that the results are not going to be sent over the network or stored in any files (save games, replays, maps etc)

Compiling out USE_GAMEMATH currently causes compile error
```
D:\Projects\TheAssemblyArmada\Thyme_Jonwil\src\w3d\math\gamemath.h(72,9): error C3861: 'gm_isnanf': identifier not found
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -248,7 +248,7 @@ if(USE_GAMEMATH)
     FetchContent_Declare(
         gamemath_git
         GIT_REPOSITORY https://github.com/TheAssemblyArmada/GameMath.git
-        GIT_TAG        f133b2cc8f85de6c2967ae77177daf0b8bbf39c0
+        GIT_TAG        59f7ccd494f7e7c916a784ac26ef266f9f09d78d
     )
 
     # We don't use FetchContent_MakeAvailable here because we don't want all gamemath targets including, just our dependencies.

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -574,7 +574,7 @@ if(USE_SDL2)
 endif()
 
 if(USE_GAMEMATH)
-    list(APPEND GAME_LINK_LIBRARIES gamemath_static_lib)
+    list(APPEND GAME_LINK_LIBRARIES gamemath::gamemath)
     list(APPEND GAME_COMPILE_OPTIONS -DBUILD_WITH_GAMEMATH)
 endif()
 

